lib/compress.js

```diff
@@ -75,18 +75,36 @@ function Compressor(options, false_by_default) {
         screw_ie8     : false,
         drop_console  : false,
         angular       : false,
-
         warnings      : true,
-        global_defs   : {}
+        global_defs   : {},
+        passes        : 1,
     }, true);
+    this.warnings_produced = {};
 };
 
 Compressor.prototype = new TreeTransformer;
 merge(Compressor.prototype, {
     option: function(key) { return this.options[key] },
-    warn: function() {
-        if (this.options.warnings)
-            AST_Node.warn.apply(AST_Node, arguments);
+    compress: function(node) {
+        var passes = +this.options.passes || 1;
+        for (var pass = 0; pass < passes && pass < 3; ++pass) {
```

Comment:
Makes sense - thanks so much.

I'm putting together a PR tackling the `side_effects` option with `passes` - since pure annotations are removed on the first pass, thus removing the benefit of multiple passes.  I'm having a hard time creating a small reproduction of code that doesn't get fully minified in one pass though - it's an interesting problem.